go-controller/pkg/util/multi_network.go

```diff
@@ -44,6 +44,12 @@ type BasicNetInfo interface {
 	Equals(BasicNetInfo) bool
 	GetNetworkScopedName(name string) string
 	RemoveNetworkScopeFromName(name string) string
+	GetNetworkScopedK8sMgmtIntfName(nodeName string) string
+	GetNetworkScopedClusterRouterName() string
+	GetNetworkScopedGWRouterName(nodeName string) string
+	GetNetworkScopedSwitchName(nodeName string) string
+	GetNetworkScopedJoinSwitchName() string
+	GetNetworkScopedExtSwitchName(nodeName string) string
```

Comment:
nit: Since these are really under under NetInfo they are suppose to be `NetworkScoped` maybe we can remove that part although I see they are used as `oc.GetNetworkScopedClusterRouteName`.

Reply:
As discussed offline, it's probably more explicit to keep them as is.  Otherwise:
```
oc.GetNetworkScopedClusterRouteName()
```
would change to:
```
oc.GetClusterRouteName()
```

@tssurya Wdyt?

Reply:
I like the NetworkScoped part

Reply:
Not at desk(will come back to this again), but in future PRs do we plan to use this from everywhere in BNC? Where its internally going to be defaultnetwork(empty) and other networks? Ny preference depends on that..
Like do we plan to keep getrouterName etc in DNC as is? And this is extra on top?